### PR TITLE
added restart instance by filter and az experiment

### LINKED ISSUE #67: Add restart ec2 instance with filter and availability zone

### PR DESCRIPTION
**Pull request by gtfisher**
Signed-off-by: gtfisher <grant@chaosiq.io>

closes #67

### CODE REVIEW

File: ec2_restart_instance_by_filter_and_az/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+## Running the Experiment
+
+### Required Parameters
+
+This experiment requires the following parameters:
+
+* Environment Variables
+  * `APPLICATION_ENTRYPOINT_URL` - Specifies the application entry point URL where your application can be reached in your environment.
+  * `INSTANCE_ID` - The id of the instance that will be stopped using a filter.
+  * `AVAILABILITY_REGION` - The availability region of the instance that will be stopped.
```
Comment (russmiles) [inline ec2_restart_instance_by_filter_and_az/README.md]: Should this ideally be "AVAILABILITY_ZONE" as that is what it usually means in AWS terminology?

File: ec2_restart_instance_by_filter_and_az/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+With the native `chaos` command:
+
+```bash
+(chaostk) export APPLICATION_ENTRYPOINT_URL=http://192.168.99.100/myapp; \
+          export INSTANCE_ID=i-1234567890abcdef0; \
+          export AVAILABILITY_REGION=eu-west-2 ; \
+          chaos run ec2_restart_instance_by_filter_and_az.json
+```
+
+***NOTE:*** The APPLICATION_ENTRYPOINT_URL and the EC2_INSTANCE_ID should be setup according to your AWS run time environment. The docker command also uses the .chaostoolkit/settings.yaml from the users home directory.
```
Comment (russmiles) [inline ec2_restart_instance_by_filter_and_az/README.md]: Note seems out of date?

File: ec2_restart_instance_by_filter_and_az/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+With the native `chaos` command:
+
+```bash
+(chaostk) export APPLICATION_ENTRYPOINT_URL=http://192.168.99.100/myapp; \
+          export INSTANCE_ID=i-1234567890abcdef0; \
+          export AVAILABILITY_REGION=eu-west-2 ; \
+          chaos run https://raw.githubusercontent.com/open-chaos/experiment-catalog/master/aws/ec2_restart_instance_by_filter_and_az/ec2_restart_instance_by_filter_and_az.json
+```
+
+***NOTE:*** The APPLICATION_ENTRYPOINT_URL and the EC2_INSTANCE_ID should be setup according to your AWS run time environment. The docker command also uses the .chaostoolkit/settings.yaml from the users home directory.
```
Comment (russmiles) [inline ec2_restart_instance_by_filter_and_az/README.md]: Note seems out of date?

File: ec2_restart_instance_by_filter_and_az/ec2_restart_instance_by_filter_and_az.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "method": [
+        {
+            "type": "action",
+            "name": "stop_instance",
+            "provider": {
+                "type": "python",
+                "module": "chaosaws.ec2.actions",
+                "func": "stop_instance",
+                "arguments": {
+                    "az": "${availability_region}",
```
Comment (russmiles) [inline ec2_restart_instance_by_filter_and_az/ec2_restart_instance_by_filter_and_az.json]: "az" is usually taken to mean "availability zone", but we seem to have gone for "region" which is unfamiliar to me. I could be wrong though?

File: ec2_restart_instance_by_filter_and_az/README.md (hunk 2 above)
Comment (gtfisher) [reply ec2_restart_instance_by_filter_and_az/README.md]: note updated

File: ec2_restart_instance_by_filter_and_az/README.md (hunk 3 above)
Comment (gtfisher) [reply ec2_restart_instance_by_filter_and_az/README.md]: note updated

**Pull request merged**